### PR TITLE
Configure quartz configuration via dropwizard

### PR DESCRIPTION
**Pull request by schmitzhermes**
Hey, as stated in #50 I provided a PR. We are going to need it anyway, so I thought I'd just give it a shot. 

Feedback is very welcome. Sorry for some formatting issues...

### CODE REVIEW

File: dropwizard-jobs-core/src/main/java/de/spinscale/dropwizard/jobs/JobManager.java [hunk 1]
```diff
@@ -218,34 +225,58 @@ protected Trigger executeNowTrigger() {
         return TriggerBuilder.newTrigger().startNow().build();
     }
 
+    private void createScheduler() throws SchedulerException {
+        // backwards compatibility - only use this when something is provided in
+        // the config
+        if (!configuration.getQuartzConfiguration().isEmpty()) {
```
Comment (tobilarscheid) [inline dropwizard-jobs-core/src/main/java/de/spinscale/dropwizard/jobs/JobManager.java]: if you put it the other way around
```java
if (configuration.getQuartzConfiguration().isEmpty()){
    scheduler = StdSchedulerFactory.getDefaultScheduler();
}
```
you avoid the double negation - I find that a lot easier to read.

**Comment by xargsgrep** [comment]
This is good, I had done the same thing by creating a `QuartzConfiguration` POJO class but that is then limited to the config properties defined in the class. Doing it this way allows configuring any of the properties without having to add them to a POJO first so I like this better.

File: dropwizard-jobs-core/src/main/java/de/spinscale/dropwizard/jobs/JobManager.java [hunk 2]
```diff
@@ -218,34 +225,58 @@ protected Trigger executeNowTrigger() {
...
                 .storeDurably(storeDurably);
 
         if (!scheduler.checkExists(jobKey)) {
-            // if the job doesn't already exist, we can create it, along with its trigger. this prevents us
-            // from creating multiple instances of the same job when running in a clustered environment
+            // if the job doesn't already exist, we can create it, along with
+            // its trigger. this prevents us
+            // from creating multiple instances of the same job when running
+            // in
+            // a clustered environment
```
Comment (xargsgrep) [inline dropwizard-jobs-core/src/main/java/de/spinscale/dropwizard/jobs/JobManager.java]: looks like the formatting in all the comments got messed up?

**Comment by schmitzhermes** [comment]
yes. we _could_ think about creating a POJO with the most used properties + the current version as a fallback. But I actually didn't feel the need to do this right now.

File: dropwizard-jobs-core/src/main/java/de/spinscale/dropwizard/jobs/JobManager.java (hunk 2 above)
Comment (schmitzhermes) [inline dropwizard-jobs-core/src/main/java/de/spinscale/dropwizard/jobs/JobManager.java]: yea - could you guys provide me your `.settings` folder in eclipse? then I'd use your code formatting settings and clean up this mess

Comment (schmitzhermes) [inline dropwizard-jobs-core/src/main/java/de/spinscale/dropwizard/jobs/JobManager.java]: ah ok, sorry I am blind. Did not see the `code-format.xml` in the repository.

**Comment by schmitzhermes** [comment]
@xargsgrep I corrected the code formatting based on the provided XML file. However, the PR still contains reformatting at certain places (e.g. new line after if statements).

I generally think it'd be preferable that we open another PR, reformat all the code in the repository based on the provided XML and merge it, so no other contributor has to worry about the code format. Moreover it'd be good if you guys provided another xml file for imports (the order depends on your settings and this settings are not included in the code format settings - only God knows why). What do you think?

**Comment by xargsgrep** [comment]
@schmitzhermes Ok great, yes I agree about the POJO - what I meant to say was that the way you have done it now is more flexible so I prefer your method over the POJO method.

Also, I just started contributing to this project very recently so I don't really know what the other people prefer in terms of code formatting and such. If others are ok with it then I'll go ahead and merge the PR.

@MusikPolice Are you ok with the changes?

**Comment by schmitzhermes** [comment]
Any updates on this?

**Comment by xargsgrep** [comment]
The changes look fine to me and I haven't heard anything from any of the others so I'm fine merging it. However, I won't be able to release to the current group ID since I don't have access to it. I'm ok doing the releases but I will have to change the group id to my own then (com.xargsgrep).

Any objections?

**Comment by x1a0** [comment]
Any plan on releasing a new version including this change to Maven?

**Pull request merged**